src/openzeppelin/upgrades/library.cairo

```diff
@@ -35,83 +35,91 @@ end
-        assert initialized = FALSE
+namespace Proxy:
+
+    #
+    # Initializer
+    #
+
+    func initializer{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(proxy_admin: felt):
+        let (initialized) = Proxy_initialized.read()
+        with_attr error_message("Proxy: contract already initialized"):
+            assert initialized = FALSE
+        end
+
+        Proxy_initialized.write(TRUE)
+        Proxy_admin.write(proxy_admin)
+        return ()
     end
 
-    Proxy_initialized.write(TRUE)
-    Proxy_admin.write(proxy_admin)
-    return ()
-end
-
-#
-# Upgrades
-#
-
-func Proxy_set_implementation{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        range_check_ptr
-    }(new_implementation: felt):
-    Proxy_implementation_address.write(new_implementation)
-    Upgraded.emit(new_implementation)
-    return ()
-end
-
-#
-# Guards
-#
+    #
+    # Upgrades
+    #
+
+    func set_implementation{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(new_implementation: felt):
+        Proxy_implementation_address.write(new_implementation)
+        Upgraded.emit(new_implementation)
+        return ()
+    end
 
-func Proxy_only_admin{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        range_check_ptr
-    }():
-    let (caller) = get_caller_address()
-    let (admin) = Proxy_admin.read()
-    with_attr error_message("Proxy: caller is not admin"):
-        assert admin = caller
+    #
+    # Setters
+    #
+
+    func set_admin{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }(new_admin: felt):
+        Proxy_admin.write(new_admin)
+        return ()
     end
-    return ()
-end
 
-#
-# Getters
-#
+    #
+    # Getters
+    #
+
+    func get_implementation{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }() -> (implementation: felt):
+        let (implementation) = Proxy_implementation_address.read()
+        return (implementation)
+    end
 
-func Proxy_get_admin{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        range_check_ptr
-    }() -> (admin: felt):
-    let (admin) = Proxy_admin.read()
-    return (admin)
-end 
-
-func Proxy_get_implementation{
-        syscall_ptr: felt*,
-        pedersen_ptr: HashBuiltin*,
-        range_check_ptr
-    }() -> (implementation: felt):
-    let (implementation) = Proxy_implementation_address.read()
-    return (implementation)
-end 
+    func get_admin{
+            syscall_ptr: felt*,
+            pedersen_ptr: HashBuiltin*,
+            range_check_ptr
+        }() -> (admin: felt):
+        let (admin) = Proxy_admin.read()
+        return (admin)
+    end
 
-#
-# Setters
-#
+    #
+    # Guards
+    #
+
+    func _only_admin{
```

Comment:
what do you think of prepending guard-like functions with `assert_`? like `assert_only_admin`. I think it's pretty clear what it does, and it would be consistent across the language and with other codebases.

Reply:
I was actually thinking that the guard functions look a little naked when they're used. Prepending with `assert_` is a really good idea